File: src/Entity/Hotel.php
<?php

namespace App\Entity;

use App\Repository\HotelRepository;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

/**
 * @ORM\Entity(repositoryClass=HotelRepository::class)
 */
class Hotel
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     */
    private $id;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="veuillez saisir le nom de l'hotel")
     *     @Assert\Length(
     *     min=6,
     *     max=15,
     *     minMessage = "le nom de l'hotel doit etre supérieur a 6 caracteres",
     *     maxMessage  ="le nom de l'hotel ne doit pas depasser 15 caracteres",
     *     allowEmptyString = false
     *     )
     * @Assert\Type(
     * type={"alpha"},
     * message="Nom Hotel  {{ value }} doit contenir seulement des lettres alphabétiques")
     */
    private $NomHotel;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="veuillez saisir l'adresse de l'hotel")
     *     @Assert\Length(
     *     min=5,
     *     max=30,
     *     minMessage = "l' Adresse de l'hotel doit etre supérieur a 5 caracteres",
     *     maxMessage  ="l' Adresse de l'hotel ne doit pas depasser 30 caracteres",
     *     allowEmptyString = false
     *     )
     */
    private $Adresse;

    /**
     * @ORM\Column(type="integer")
     * @Assert\NotBlank(message="veuillez saisir le numéro de téléphone")
     *      @Assert\Length(
     *      min=8,
     *      max=8,
     *      minMessage = "Le numéro de téléphone doit etre composé de 8 chiffres",
     *      maxMessage  ="Le numéro de téléphone doit etre composé de 8 chiffres",
     * )
     */
    private $NumTel;

    /**
     * @ORM\Column(type="integer")
     * @Assert\NotBlank(message="veuillez saisir le nombre d'étoiles")
     */
    private $NbrEtoiles;

    /**
     * @ORM\Column(type="integer")
     * @Assert\NotBlank(message="veuillez saisir le nombre des chambres")
     * @Assert\GreaterThanOrEqual(
     *     value = 50,message="Le nombre minimale de chambres est = 50"
     * )
     * @Assert\LessThanOrEqual(
     *     value = 1000,message="Le nombre de chambres ne doit pas depasser 1000"
     * )
     */
    private $NbrChambre;

    /**
     * @ORM\Column(type="integer")
     * @Assert\NotBlank(message="veuillez saisir le code hotel")
     */
    private $CodeH;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="ce champs est obligatoire")
     * @Assert\Email(message = " cette adresse '{{ value }}' est invalide .")
     */
    private $email;
    

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNomHotel(): ?string
    {
        return $this->NomHotel;
    }

    public function setNomHotel(string $NomHotel): self
    {
        $this->NomHotel = $NomHotel;

        return $this;
    }

    public function getAdresse(): ?string
    {
        return $this->Adresse;
    }

    public function setAdresse(string $Adresse): self
    {
        $this->Adresse = $Adresse;

        return $this;
    }

    public function getNumTel(): ?int
    {
        return $this->NumTel;
    }

    public function setNumTel(int $NumTel): self
    {
        $this->NumTel = $NumTel;

        return $this;
    }

    public function getNbrEtoiles(): ?int
    {
        return $this->NbrEtoiles;
    }

    public function setNbrEtoiles(int $NbrEtoiles): self
    {
        $this->NbrEtoiles = $NbrEtoiles;

        return $this;
    }

    public function getNbrChambre(): ?int
    {
        return $this->NbrChambre;
    }

    public function setNbrChambre(int $NbrChambre): self
    {
        $this->NbrChambre = $NbrChambre;

        return $this;
    }

    public function getCodeH(): ?int
    {
        return $this->CodeH;
    }

    public function setCodeH(int $CodeH): self
    {
        $this->CodeH = $CodeH;

        return $this;
    }

    public function getEmail(): ?string
    {
        return $this->email;
    }

    public function setEmail(string $email): self
    {
        $this->email = $email;

        return $this;
    }


}

File: src/entities/Hotel.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package entities;

import java.util.Objects;

/**
 *
 * @author 123
 */
public class Hotel implements Comparable<Hotel> {

    private int id, NumTel, NbrEtoiles, NbrChambre, CodeH;
    private String NomHotel, Adresse, email, imageHotel;

    public Hotel() {
    }

    public Hotel(int NumTel, int NbrEtoiles, int NbrChambre, int CodeH, String NomHotel, String Adresse, String email, String imageHotel) {
        this.NumTel = NumTel;
        this.NbrEtoiles = NbrEtoiles;
        this.NbrChambre = NbrChambre;
        this.CodeH = CodeH;
        this.NomHotel = NomHotel;
        this.Adresse = Adresse;
        this.email = email;
        this.imageHotel = imageHotel;
    }

    public Hotel(int id, int NumTel, int NbrEtoiles, int NbrChambre, int CodeH, String NomHotel, String Adresse, String email, String imageHotel) {

        this.id = id;
        this.NumTel = NumTel;
        this.NbrEtoiles = NbrEtoiles;
        this.NbrChambre = NbrChambre;
        this.CodeH = CodeH;
        this.NomHotel = NomHotel;
        this.Adresse = Adresse;
        this.email = email;
        this.imageHotel = imageHotel;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getNumTel() {
        return NumTel;
    }

    public void setNumTel(int NumTel) {
        this.NumTel = NumTel;
    }

    public int getNbrEtoiles() {
        return NbrEtoiles;
    }

    public void setNbrEtoiles(int NbrEtoiles) {
        this.NbrEtoiles = NbrEtoiles;
    }

    public int getNbrChambre() {
        return NbrChambre;
    }

    public void setNbrChambre(int NbrChambre) {
        this.NbrChambre = NbrChambre;
    }

    public int getCodeH() {
        return CodeH;
    }

    public void setCodeH(int CodeH) {
        this.CodeH = CodeH;
    }

    public String getNomHotel() {
        return NomHotel;
    }

    public void setNomHotel(String NomHotel) {
        this.NomHotel = NomHotel;
    }

    public String getAdresse() {
        return Adresse;
    }

    public void setAdresse(String Adresse) {
        this.Adresse = Adresse;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getImageHotel() {
        return imageHotel;
    }

    public void setImageHotel(String imageHotel) {
        this.imageHotel = imageHotel;
    }

    @Override
    public String toString() {
        return "\nHotel{" + "id=" + id + ", NumTel=" + NumTel + ", NbrEtoiles=" + NbrEtoiles + ", NbrChambre=" + NbrChambre + ", CodeH=" + CodeH + ", NomHotel=" + NomHotel + ", Adresse=" + Adresse + ", email=" + email + ", imageHotel=" + imageHotel + '}';
    }

    @Override
    public int hashCode() {
        int hash = 7;
        return hash;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (obj == null) {
            return false;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }
        final Hotel other = (Hotel) obj;
        if (this.NumTel != other.NumTel) {
            return false;
        }
        if (this.NbrEtoiles != other.NbrEtoiles) {
            return false;
        }
        if (this.NbrChambre != other.NbrChambre) {
            return false;
        }
        if (this.CodeH != other.CodeH) {
            return false;
        }
        if (!Objects.equals(this.NomHotel, other.NomHotel)) {
            return false;
        }
        if (!Objects.equals(this.Adresse, other.Adresse)) {
            return false;
        }
        if (!Objects.equals(this.email, other.email)) {
            return false;
        }
        if (!Objects.equals(this.imageHotel, other.imageHotel)) {
            return false;
        }
        return true;
    }

    @Override
    public int compareTo(Hotel o) {

        if ((this.CodeH - o.CodeH) != 0) {
            return this.CodeH - o.CodeH;
        } 
        else if (this.NomHotel.compareTo(o.NomHotel)!=0)
        {
            return this.NomHotel.compareTo(o.NomHotel);
        }
        else if (this.Adresse.compareTo(o.Adresse)!=0)
        {
            return this.Adresse.compareTo(o.Adresse);
        }
        else if ((this.NbrEtoiles - o.NbrEtoiles) != 0)
        {
            return this.NbrEtoiles - o.NbrEtoiles;
        }
        else if ((this.NbrChambre - o.NbrChambre) != 0)
        {
            return this.NbrChambre - o.NbrChambre;
        }
        else if ((this.NumTel - o.NumTel) != 0)
        {
            return this.NumTel - o.NumTel;
        }
        else 
           return this.id - o.id;
    }

}
